JSDemos/Demos/TreeView/TreeViewWithSearchBar/description.md

```diff
@@ -1 +1,11 @@
-This demo demonstrates the TreeView component&rsquo;s searching capabilities. The component can find nodes containing your search string or&nbsp;only starting with it&nbsp;depending on&nbsp;the search mode.
\ No newline at end of file
+To add a search bar to the TreeView and customize the search functionality, do the following:
+
+- Set the [searchEnabled](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchEnabled) property to **true**.
+
+- Use the [searchMode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchMode) property to specify whether items should contain (default), start with, or match the search string. In this example, you can change the **searchMode** in the drop-down menu.
+
+- Configure the [searchExpr](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchExpr) property to specify custom search data fields. The default search field is [text](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/items/#text). 
+
+The TreeView uses the TextBox component as the search bar. To customize it, specify [TextBox properties](/Documentation/ApiReference/UI_Components/dxTextBox/Configuration/) in the [searchEditorOptions](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#searchEditorOptions) object.
+
+We do not recommend that you enable search if the TreeView uses [virtual mode](https://js.devexpress.com/Demos/WidgetsGallery/Demo/TreeView/VirtualMode/) or [custom logic](https://js.devexpress.com/Demos/WidgetsGallery/Demo/TreeView/LoadDataOnDemand/) to load data on demand. In these cases, the TreeView may not have the full dataset, and the search results include only loaded nodes.
```

Comment:
You can percieve the active mode as a state of the TreeView control. It does not use the mode, it can either be in this mode or have this mode as active. The point is that we do not recommend customers to enable the search functionality if the TreeView loads data on demand for whatever reason.

> In these cases, the TreeView may not have the full dataset, and the search results include only loaded nodes.

I'd like to draw your attention towards this sentence because it suggests that the TreeView may or may not have the full dataset at its disposal.  However, this is not quite true, because the TreeView will not have the full dataset in 99,9% of cases.

One more thing, why don't we recommend them to enable the search functionality? Are the search results incorrect? No. They are correct. However, they only reflect a subset, a small portion of the search results we would have if we'd loaded the full dataset in advance.

I would have rewritten this paragraph and would have said that recommend you to use search functionality with caution when the TreeView control loads data on demand which is the case when virtual mode is active or you employ custom data load algorithms.